tests/impls/image_reader/test_gdal_io.py

```diff
@@ -253,6 +236,26 @@ class TestGdalImageReader (unittest.TestCase):
 
     If this was not skipped, ``osgeo`` must have been importable.
     """
+    gh_image_fp: str
+    gh_file_element: DataFileElement
+    gh_cropped_image_fp: str
+    gh_cropped_file_element: DataFileElement
+    gh_cropped_bbox: AxisAlignedBoundingBox
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        # Initialize test image paths/elements/associated crop boxes.
+
+        cls.gh_image_fp = os.path.join(TEST_DATA_DIR, "grace_hopper.png")
+        cls.gh_file_element = DataFileElement(cls.gh_image_fp, readonly=True)
```

Comment:
would this extend to `test_pil_io.py` as well?

Reply:
If the PIL unittests uses temp file writing, it should clean those up (I don't recall specifics in there at the moment).

This was more particularly relevant here because there *was* clean-up happening when the methods were at the module level, but they disapeared when you moved the equivalent functionality into the class level.

Reply:
Granted, a temp-file cleanup call is not going to hurt anything as I currently see you have it written.\